src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -67,7 +67,7 @@ public override IEnumerable<string> PreprocessorSymbolNames
             this.SpecifiedLanguageVersion = languageVersion;
             this.LanguageVersion = languageVersion.MapSpecifiedToEffectiveVersion();
             this.PreprocessorSymbols = preprocessorSymbols.ToImmutableArrayOrEmpty();
-            _features = features?.ToImmutableDictionary() ?? ImmutableDictionary<string, string>.Empty;
+            _features = features?.ToImmutableDictionary(StringComparer.OrdinalIgnoreCase) ?? ImmutableDictionary<string, string>.Empty;
```

Comment:
>  Our suspicion is the root cause lies in a culture-specific manipulation of case somewhere prior to the use of an ordinal comparison in this method, but to date no one has conclusively identified any such location.

It sounds like we are trying to fix a symptom rather than the cause.